connect-inject/container_init_test.go

```diff
@@ -247,6 +325,7 @@ export CONSUL_HTTP_ADDR="${HOST_IP}:8500"
 export CONSUL_GRPC_ADDR="${HOST_IP}:8502"
 consul-k8s connect-init -pod-name=${POD_NAME} -pod-namespace=${POD_NAMESPACE} \
   -acl-auth-method="auth-method" \
+  -service-account-name="" \
```

Comment:
scratching my head on this one a little bit.
Isn't it invalid to have auth-method and no service-account-name? Shouldnt we always pass it because we'll always know it at this point?

Reply:
I needed a way to know whether to compare the service account name to the kubernetes service name or not, since we only want to compare them if the annotation is not set. I had originally done that with a separate flag like `-service-name-annotation-set` but had the idea to consolidate to one flag, just `-service-account-name` and do the check based on whether the flag had a value or not. This was essentially the way to signal from container_init.go to the connect-init command to do the check or not.

Reply:
If we always set the service account name, it will always check it against the kube service name which is undesirable if the service name annotation has been set to be different from the kube service name.

Reply:
Ah right I'm remembering now, okay that makes sense, thanks!